### PR TITLE
fix #2310 useWatch with useFieldArray delete

### LINKED ISSUE #2310: useWatch value become empty after delete field array item
**Describe the bug**

**To Reproduce**
1. Go to https://codesandbox.io/s/react-hook-form-v6-usewatch-mjnow?file=/src/index.js
2. Click on append
3. Click on append
4. Click on delete
5. See JSON output

**Expected behavior**

it should preserve value same as `watch`

**Screenshots**
image
image

### PR DESCRIPTION
**Pull request by bluebill1049**
- [x] unit test

close #2310

### CODE REVIEW

File: src/useFieldArray.ts [hunk 1]
```diff
@@ -443,12 +443,16 @@ export const useFieldArray = <
     if (isWatchAllRef.current) {
       reRender();
     } else if (watchFieldsRef) {
+      let shouldRenderUseWatch = true;
       for (const watchField of watchFieldsRef.current) {
         if (watchField.startsWith(name)) {
           reRender();
+          shouldRenderUseWatch = false;
```
Comment (keiya01) [inline src/useFieldArray.ts:450]: Isn't this extra `re-render` ?

Comment (bluebill1049) [reply src/useFieldArray.ts:450]: reRender will trigger the re-render at the root (useForm) level, so we probably can skip the extra re-render for `useWatch`.

Comment (keiya01) [reply src/useFieldArray.ts:450]: Oh I see. I misunderstood ...

**Review by keiya01 — APPROVED** [review]
LGTM :)

File: src/useFieldArray.ts (hunk 1 above)
Comment (bluebill1049) [reply src/useFieldArray.ts:450]: thanks for the detailed review @keiya01 🙏

**Pull request merged**